### PR TITLE
Improve error handling when invalid report is entered

### PR DESCRIPTION
**Pull request by taers232c**
Replace this:
```
$ gam report foo
Traceback (most recent call last):
  File "/Users/admin/Documents/GoogleApps/GAMO/gam.py", line 14755, in <module>
    sys.exit(ProcessGAMCommand(sys.argv))
  File "/Users/admin/Documents/GoogleApps/GAMO/gam.py", line 14391, in ProcessGAMCommand
    showReport()
  File "/Users/admin/Documents/GoogleApps/GAMO/gam.py", line 1386, in showReport
    activities = gapi.get_all_pages(rep.activities(), 'list', 'items',
  File "/Users/admin/Documents/GoogleApps/GAMO/gapi/__init__.py", line 254, in get_all_pages
    page_key = _get_max_page_size_for_api_call(service, function, **kwargs)
  File "/Users/admin/Documents/GoogleApps/GAMO/gapi/__init__.py", line 157, in _get_max_page_size_for_api_call
    api_id = method(**kwargs).methodId
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/googleapiclient/discovery.py", line 741, in method
    raise TypeError(
TypeError: Parameter "applicationName" value "foo" does not match the pattern "(admin)|(calendar)|(drive)|(login)|(mobile)|(token)|(groups)|(saml)|(chat)|(gplus)|(rules)|(jamboard)|(meet)|(user_accounts)|(access_transparency)|(groups_enterprise)|(gcp)"
```
with this:
```
$ gam report foo

ERROR: report must be one of access, accesstransparency, admin, calendar, calendars, customer, customers, doc, docs, domain, drive, enterprisegroups, gcp, gplus, google+, group, groups, groupsenterprise, hangoutsmeet, jamboard, login, logins, meet, mobile,\
 oauthtoken, rules, saml, token, tokens, user, users, useraccounts; got foo

```

### CODE REVIEW

**Comment by jay0lee** [comment]
The discovery doc at:

https://www.googleapis.com/discovery/v1/apis/admin/reports_v1/rest

enumerates the allowed reports. Let's use that instead of a static list so as new reports are added by Google they just work with GAM.

File: src/gam.py [hunk 1]
```diff
@@ -1213,9 +1213,34 @@ def _checkFullDataAvailable(warnings, tryDate, fullDataRequired):
...
+  'tokens': 'token',
+  'users': 'user',
+  'useraccounts': 'user_accounts',
+  }
+
 def showReport():
   rep = buildGAPIObject('reports')
   report = sys.argv[2].lower()
+  if report.replace('_', '') in REPORT_CHOICE_MAP:
+    report = REPORT_CHOICE_MAP[report.replace('_', '')]
```
Comment (jay0lee) [inline src/gam.py]: simplify to:
```
report = sys.argv[2].lower().replace('_', '')
report  = REPORT_CHOICE_MAP.get(report, report)

Comment (taers232c) [reply src/gam.py]: If I make this change, then a new app foo_bar in the discovery doc can't be recognized as we stripped out the _ and don't have a map to put it back

Comment (taers232c) [reply src/gam.py]: This will work, just posted:
  report = sys.argv[2].lower()
  report = REPORT_CHOICE_MAP.get(report.replace('_', ''), report)

**Pull request merged**